src/workerd/io/io-context.h

```diff
@@ -744,6 +744,13 @@ class IoContext final: public kj::Refcounted, private kj::TaskSet::ErrorHandler
       kj::ConstString operationName,
       std::initializer_list<SpanTagParams> tags);
 
+  // As above, but with callback to add span tags to the trace context for the operation.
+  kj::Own<kj::HttpClient> getHttpClientWithCallback(uint channel,
```

Comment:
not very big fan of callback version here.  why can't we make `WithSpans` work for us? (or maybe extend it?)

Reply:
A couple of thoughts here:
- We can't use the existing WithSpans() here since it takes an initializer list. Initializer lists are designed as a very lightweight interface, we can't append to them which makes them very inconvenient when there is a variable number of span tags.
- I pushed a PR that adds a WithSpans() version that takes a kj::Vector. The current approach is templated so that we still have the initializer list version available (which has less overhead, more syntactic sugar potential), but it looks like we might not even need the old version. Will push another change...

Reply:
Updated, much cleaner now and not using a callback.